mentat/edit_history.py

```diff
@@ -1,71 +1,88 @@
 import os
 from pathlib import Path
+from typing import Optional
 
 import attr
 from termcolor import colored
 
+from mentat.errors import MentatError
+from mentat.parsers.file_edit import FileEdit, Replacement
+from mentat.session_context import SESSION_CONTEXT
+
 
 # All paths should be abs paths
 @attr.define()
 class RenameAction:
     old_file_name: Path = attr.field()
     cur_file_name: Path = attr.field()
 
-    def undo(self) -> str | None:
+    def undo(self) -> FileEdit:
         if self.old_file_name.exists():
-            return colored(
+            raise MentatError(
```

Comment:
We don't want to end the session when an undo fails, which is what throwing a MentatError will do